app/scripts/controllers/app-state.js

```diff
@@ -374,6 +374,25 @@ export default class AppStateController extends EventEmitter {
     this.store.updateState({ usedNetworks });
   }
 
+  ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
+  /**
+   * Set the interactive replacement token with a url and the old refresh token
+   *
+   * @param {object} opts
+   * @param opts.url
+   * @param opts.oldRefreshToken
+   * @returns {void}
+   */
+  showInteractiveReplacementTokenBanner({ url, oldRefreshToken }) {
+    this.store.updateState({
+      interactiveReplacementToken: {
```

Comment:
What is this state responsible for? I'm not familiar with interactive replacement tokens.

Reply:
This allows users to replace expired custodian refresh tokens automatically. When `interactiveReplacementToken` has the `url` and the `oldRefreshToken`, we show that little banner on top of the extension. More info: [url](https://consensyssoftware.atlassian.net/wiki/spaces/PRDC/pages/398657881166/05.+Token+replacement)

![image](https://github.com/MetaMask/metamask-extension/assets/1182864/6106a9f4-6a48-4fb0-925b-05c7e7a1c959)

Reply:
Makes sense, thanks!